src/Symfony/Bridge/Monolog/Processor/DebugProcessor.php

```diff
@@ -31,37 +29,29 @@ public function __construct(RequestStack $requestStack = null)
         $this->requestStack = $requestStack;
     }
 
-    private function doInvoke(array|LogRecord $record): array|LogRecord
+    public function __invoke(LogRecord $record): LogRecord
     {
         $key = $this->requestStack && ($request = $this->requestStack->getCurrentRequest()) ? spl_object_id($request) : '';
 
-        $timestampRfc3339 = false;
-        if ($record['datetime'] instanceof \DateTimeInterface) {
-            $timestamp = $record['datetime']->getTimestamp();
-            $timestampRfc3339 = $record['datetime']->format(\DateTimeInterface::RFC3339_EXTENDED);
-        } elseif (false !== $timestamp = strtotime($record['datetime'])) {
-            $timestampRfc3339 = (new \DateTimeImmutable($record['datetime']))->format(\DateTimeInterface::RFC3339_EXTENDED);
-        }
-
         $this->records[$key][] = [
-            'timestamp' => $timestamp,
-            'timestamp_rfc3339' => $timestampRfc3339,
-            'message' => $record['message'],
-            'priority' => $record['level'],
-            'priorityName' => $record['level_name'],
-            'context' => $record['context'],
-            'channel' => $record['channel'] ?? '',
+            'timestamp' => $record->datetime->getTimestamp(),
+            'timestamp_rfc3339' => $record->datetime->format(\DateTimeInterface::RFC3339_EXTENDED),
+            'message' => $record->message,
+            'priority' => $record->level->value,
+            'priorityName' => $record->level->getName(),
+            'context' => $record->context,
+            'channel' => $record->channel ?? '',
         ];
 
         if (!isset($this->errorCount[$key])) {
             $this->errorCount[$key] = 0;
         }
 
-        switch ($record['level']) {
-            case Logger::ERROR:
-            case Logger::CRITICAL:
-            case Logger::ALERT:
-            case Logger::EMERGENCY:
+        switch ($record->level) {
+            case Level::Error:
+            case Level::Critical:
+            case Level::Alert:
+            case Level::Emergency:
                 ++$this->errorCount[$key];
         }
```

Comment:
Could also be like this AFAICT:

```suggestion
        if ($record->level->isHigherThan(Level::Warning)) {
            ++$this->errorCount[$key];
        }
```